File: problem/bry/Kotlin/shuffle-the-array.kt
class Solution {
    fun shuffle(nums: IntArray, n: Int) = IntArray(n*2).apply {
        (0 until n).forEach {
            set(it.shl(1), nums.get(it))
            set(it.shl(1)+1, nums.get(it + n))
        }
    }
}

//Special save for Kotlin

File: problem/bry/C++/shuffle-the-array.cpp
class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        vector<int> ans(n<<1);
        for(int i = 0; i < n; ++i){
            ans[(i<<1)] = (nums[i]);
            ans[(i<<1)+1]= (nums[i+n]);
        }
        return ans;
    }
};